src/skill.cpp

```diff
@@ -139,7 +139,8 @@ void Skill::load_skill( const JsonObject &jsobj )
         jso_tta.read( "time_reduction_per_level", time_to_attack.time_reduction_per_level );
     }
     skill_displayType_id display_type = skill_displayType_id( jsobj.get_string( "display_category" ) );
-    Skill sk( ident, name, desc, jsobj.get_tags( "tags" ), display_type );
+    int ordering = jsobj.get_int( "ordering", 100000000 );
+    Skill sk( ident, name, desc, jsobj.get_tags( "tags" ), display_type, ordering );
```

Comment:
You don't need any of the constructor changes, especially if you just feed it garbage value as a default anyway.

If you remove constructor changes you can just assign it like below.

Also a skippable alert so out-of-tree mod authors are aware they need to update their json would be useful because the garbage value will make multiple skills with the garbage value have undefined order, something similar to:
```suggestion
    Skill sk( ident, name, desc, jsobj.get_tags( "tags" ), display_type );
    if( jsobj.has_int( "ordering" ) ) {
        sk.ordering = jsobj.get_int( "ordering" );
    } else {
        sk.ordering = 1000000;
        debugmsg( "skill '%s' missing 'ordering' field.", ident.str() );
    }
```

The PR also misses updating in-tree mod skills - aftershock has smart guns and magiclysm has spellcraft skills, may be more

Also minor nitpick; "ordering" doesn't fit mostly-noun naming scheme of json fields

Reply:
I've applied all changes except the renaming of the "ordering" field because I can't think of something better.  I'll change it to whatever you guys will tell me :)

Reply:
following andrei8l suggested term, I've changed "ordering" into "sort_rank"